### DISCUSSION TITLE
Taking a look at solver discrepancies

### DISCUSSION THREAD

**Discussion by microprediction** [qpsolvers/qpsolvers#184] [General]
I've been using your package (thanks) to look at some fin-related problems, where the solvers disagree with default settings. 

- Here's a [colab notebook](https://github.com/microprediction/qpsolversbench/blob/main/examples/fin_discrepancy.ipynb) that generates a correlation matrix (of the largest discrepancy coordinate) as a diagnostic.
 
- I shoved a few utils in qpsolversbench since I didn't want to introduce a dependency. 

I am wondering if solution differences can be solved by from a tolerance setting somewhere.  Its interesting that the interior point methods agree versus the aug Lagrange.

**Comment by stephane-caron** [comment]
Thanks for your feedback!

> I am wondering if solution differences can be solved by from a tolerance setting somewhere.

Absolutely. As you have noticed, default settings vary widely from solver to solver, some of them are configured to be faster but less precise by default (or not provide as strong guarantees, for instance SCS has stronger guarantees than OSQP). You can take a look at [qpbenchmark](https://github.com/qpsolvers/qpbenchmark) where we define settings as consistent as possible between solvers. There is a [settings recap table](https://github.com/stephane-caron/qpsolvers_benchmark/blob/main/results/maros_meszaros.md#settings) at the beginning of each test-set report.

Hoping this helps! And if you feel so inclined you are very welcome to join in the development of this benchmark, [discussions](https://github.com/stephane-caron/qpsolvers_benchmark/discussions) on evaluating solvers are open (and everyone can [submit problems representing their use cases](https://github.com/stephane-caron/qpsolvers_benchmark/issues/new?template=new_problem.md)).

**Comment by microprediction** [comment]
Yas I was thinking to include some, but also figured you would not want dependency on precise. It has been on my list to create a tiny little package just for cov matrix generation, and that could feed into tests.

**Comment by stephane-caron** [comment]
> Yas I was thinking to include some, but also figured you would not want dependency on precise.

You're right, I should actually separate test set generation so that the benchmark itself does not depend on the software used to generate problems.

I'll do that in https://github.com/stephane-caron/qpsolvers_benchmark/issues/44, then we can add problems generated with ``precise`` to the benchmark.

**Reply by stephane-caron** [reply]
Following up, this has been implemented in https://github.com/qpsolvers/qpsolvers_benchmark/pull/73 and is now possible as of v1.0.0 of the benchmark. Everyone is welcome to bring their own dependencies into new test sets!

**Comment by microprediction** [comment]
Makes sense!

**Comment by microprediction** [comment]
As noted I'll try to slim down the precise dep so it doesn't bloat qpsolvers_benchmark and migrate qpsolversbench into there. I assume it is okay if there are some stochastically generated problems.

**Comment by stephane-caron** [comment]
Moving this thread to Discussions as follow-up tasks have been spun into dedicated issues.
